root/hist/hist/RHistFillBenchmarks.cxx

```diff
@@ -31,34 +30,105 @@ static void ConcurrentFillArguments(benchmark::internal::Benchmark *b)
+
+static void FillHist(int nbOfDataPoints, ROOT::Experimental::RH2D * hist)
+{
+   // Fill without weight
+   for (int i = 0; i < nbOfDataPoints; ++i) {
+      double d = static_cast< double >(i);
+      hist->Fill({d / nbOfDataPoints, d / (nbOfDataPoints / 10)});
+   }
+   // Fill with weights
+   for (int i = 0; i < nbOfDataPoints; ++i) {
+      double d = static_cast< double >(i);
+      hist->Fill({d / nbOfDataPoints, d / (nbOfDataPoints / 10)}, d);
+   }
+}
 
 // Benchmark for simple Fill on 2D hist
 static void BM_RHist_Fill(benchmark::State &state)
 {
+   int nbOfDataPoints = state.range(0);
    int nbOfBinsForFirstAxis = state.range(1);
    int nbOfBinsForSecondAxis = state.range(2);
 
-   ROOT::Experimental::RH2D hist{{nbOfBinsForFirstAxis, 0., 1.}, {nbOfBinsForSecondAxis, 0., 10.}};
-
-   int nbOfDataPoints = state.range(0); 
-   for (auto _ : state) {
-      // Fill without weight
-      for (int i = 0; i < nbOfDataPoints; ++i) {
-         float d = static_cast< float >(i);
-         hist.Fill({d / nbOfDataPoints, d / (nbOfDataPoints / 10)});
-      }
-
-      // Fill with weights
-      for (int i = 0; i < nbOfDataPoints; ++i) {
-         float d = static_cast< float >(i);
-         hist.Fill({d / nbOfDataPoints, d / (nbOfDataPoints / 10)}, d);
-      }
-   }
+   for (auto _ : state)
+      FillHist(nbOfDataPoints, histFill[nbOfBinsForFirstAxis][nbOfBinsForSecondAxis]);
 }
 BENCHMARK(BM_RHist_Fill) -> Apply(FillArguments);
 
+// Histograms with different bin configurations for FillN
+ROOT::Experimental::RH2D hist11{{10, 0.1, 1.}, {10 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist12{{10, 0.1, 1.}, {100 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist13{{10, 0.1, 1.}, {1000 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist14{{100, 0.1, 1.}, {10 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist15{{100, 0.1, 1.}, {100 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist16{{100, 0.1, 1.}, {1000 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist17{{1000, 0.1, 1.}, {10 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist18{{1000, 0.1, 1.}, {100 / 2, 0., 10.}};
+ROOT::Experimental::RH2D hist19{{1000, 0.1, 1.}, {1000 / 2, 0., 10.}};
+
+std::map<int, std::map<int, ROOT::Experimental::RH2D*>> histFillN = { { 10, { { 5, &hist11 }, { 50, &hist12 }, { 500, &hist13} } },
+                                    { 100, { { 5, &hist14 }, { 50, &hist15 }, { 500, &hist16 } } },
+                                    { 1000, { { 5, &hist17 }, { 50, &hist18 }, { 500, &hist19 } } } };
+
+// Benchmark for simple FillN on 2D hist
+static void BM_RHist_FillN(benchmark::State &state)
+{
+   int nbOfDataPoints = state.range(0);
+   int nbOfBinsForFirstAxis = state.range(1);
+   int nbOfBinsForSecondAxis = state.range(2);
 
+   for (auto _ : state) {
+      // FillN without weight
+      histFillN[nbOfBinsForFirstAxis][nbOfBinsForSecondAxis]->FillN(std::span<const CoordArray_t>(&coords[0], nbOfDataPoints), std::span<const double>(&weights[0], nbOfDataPoints));
```

Comment:
See comment at https://github.com/root-project/rootbench/pull/109#pullrequestreview-368793083 - that's applicable here, too.